### ISSUE TITLE
[BUG] inconsistent suffixes in console reporter

### ISSUE THREAD

**Issue by cwpearson** [google/benchmark#1009]
**Describe the bug**

for user counters, `k` means 1000 and for the `bytes_per_second` counter, `k` means 1024.

**System**
Which OS, compiler, and compiler version are you using:
  - OS: Ubuntu 18.04
  - Compiler and version: g++ 7.5.0

**To reproduce**
1. sync to tag 1.5.1
2. Create a benchmark that uses `state.SetBytesProcessed(...)` and `state.counters["bytes"] = ...`.
3. Compare the results of the csv reporter and the console reporter for the `bytes_per_second` and `bytes` counters.

**Expected behavior**

Perhaps these suffixes should have the same meaning, or the power-of-2 versions should change to `ki`, `Mi`, `Gi`.

**Additional context**

The csv output format prints the true values, for reference, so it's easy to see that the suffixes have different meanings for different user counters.

**Comment by dominichamon** [comment]
The `false` on [this line](https://github.com/google/benchmark/blob/56898e9a92fba537671d5462df9c5ef2ea6a823a/src/string_util.cc#L109) could be changed to 'true' depending on `one_k`. However, i'd also appreciate a more complete refactor that changes the double `one_k` to be an enum SI vs IEC which can be plumbed through correctly.

**Comment by varshneydevansh** [comment]
Hi @dmah42, if this is still pending, I would love to work on this one? I did look at the [code file.](https://github.com/google/benchmark/blob/56898e9a92fba537671d5462df9c5ef2ea6a823a/src/string_util.cc#L109)

It will help me gain a more profound understanding of benchmarking.

Regards.

**Comment by LebedevRI** [comment]
I'm not exactly sure how it should work (as opposed to how it works now;
should said counters be fully Unitful/dimension-ful?),
but this does seem like it should be reasonably simple changes.
I don't think anyone is looking into this currently, so feel free to.
Thanks.

**Comment by dmah42** [comment]
we do have the ability to differentiate between SI and IEC but it only allows a single character suffix.  might be good to expand this to multi-character strings and correctly differentiate in the output.

```
// kilo, Mega, Giga, Tera, Peta, Exa, Zetta, Yotta.
const char kBigSIUnits[] = "kMGTPEZY";
// Kibi, Mebi, Gibi, Tebi, Pebi, Exbi, Zebi, Yobi.
const char kBigIECUnits[] = "KMGTPEZY";
```

**Comment by dmah42** [comment]
yeah, and then picking the right one for user counters vs other reports.  in addition we might offer the user counter to be SI or IEC (and then do the calculation and reporting appropriately) but that's probably a second phase.

**Comment by dmah42** [comment]
`OneK` is already an enum and included in the `Counter`, so i don't think you need to change anything there. `HumanReadableNumber` is already passed oneK from the `Counter` so i don't think you need to change that either.

i think first changing the units to differentiate correctly between SI and IEC (they're currently the same for MGT, etc, and ambiguous for k/K) would get us a long way.

**Comment by qorexdevs** [comment]
I checked current main while looking at this issue. It looks like this is already covered now: `HumanReadableNumber()` switches to `Ki`/`Mi`/... for `Counter::kIs1024`, and `test/user_counters_thousands_test.cc` expects `976.56[23]Ki` and `1Mi` in console output.

I did not find an open PR needed here unless there is another console path still producing ambiguous suffixes.

**Comment by LebedevRI** [comment]
I'm pretty sure https://github.com/google/benchmark/pull/1631 meant to close this issue.
Thank you!

**Comment by qorexdevs** [comment]
Thanks, that matches what I saw on current main. #1631 covers the console suffix path, so closing this looks right.
